config/reader.go

```diff
@@ -384,8 +393,13 @@ func fromReader(
 		return nil, errors.Wrapf(err, "failed to process Config")
 	}
 
-	if shouldReadFromCloud && cfgFromDisk.Cloud != nil {
-		cfg, err := readFromCloud(ctx, cfgFromDisk, nil, true, true, logger)
+	if rrFactory != nil && cfgFromDisk.Cloud != nil {
+		rr, closeFunc, err := rrFactory(ctx, cfgFromDisk.Cloud, logger)
+		if err != nil {
+			return nil, err
+		}
+		defer utils.UncheckedErrorFunc(closeFunc)
+		cfg, err := rr.readFromCloud(ctx, cfgFromDisk, nil, true, true, logger)
 		return cfg, err
```

Comment:
this is the injection point. 

having a factory argument is ugly... basically we need to read some information from the config on disk to create a grpc connection to app (robot part id, secret, app address), and that reading happens here. 

curious if folks have ideas to improve this.